x-pack/solutions/observability/packages/kbn-observability-schema/related_dashboards/schema/recommended_dashboard/v1.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod';
+import { relevantPanelSchema } from '../relevant_panel/latest';
+
+export const relatedDashboardSchema = z.object({
+  id: z.string(),
+  title: z.string(),
+  matchedBy: z.object({
+    fields: z.array(z.string()).optional(),
+    index: z.array(z.string()).optional(),
+  }),
+  relevantPanelCount: z.number(),
+  relevantPanels: z.array(relevantPanelSchema),
+});
```

Comment:
❔ Should we add about some pagination schema?

Reply:
I think I might hold on this and implement in sorting issue as well. What do you think?

Because our results will be sorted by relevance, we won't be able to simply pass the pagination logic to the content client (dashboard client), so there will be some trickiness there needing specific attention.

Thoughts? Are you comfortable with that?

Reply:
Sure since the api is internal, we can break it without handling backward compatibility